pkg/operator/hive/hive.go

```diff
@@ -143,37 +154,45 @@ func (r *ReconcileHiveConfig) deployHive(hLog log.FieldLogger, h *resource.Helpe
 		hiveDeployment.Spec.Replicas = &replicas
 	}
 
-	result, err := h.ApplyRuntimeObject(hiveDeployment, scheme.Scheme)
-	if err != nil {
-		hLog.WithError(err).Error("error applying deployment")
-		return err
+	// Load namespaced assets, decode them, set to our target namespace, and apply:
+	coreCodecs := serializer.NewCodecFactory(scheme.Scheme)
+	coreDecoder := coreCodecs.UniversalDecoder(corev1.SchemeGroupVersion)
```

Comment:
I don't think that a decoder is what you want to use. I think you want to use a deserializer instead. See https://github.com/kubernetes/apimachinery/blob/591a38b7a7b7e4f327f8bdbb32dfb8f0247e411e/pkg/runtime/serializer/codec_factory.go#L253-L255.